controllers/controllers/services/bindings/upsi/controller.go

```diff
@@ -111,10 +123,86 @@ func (r *UPSIBindingReconciler) reconcileCredentials(ctx context.Context, cfServ
 		return controllerutil.SetControllerReference(cfServiceBinding, bindingSecret, r.scheme)
 	})
 	if err != nil {
-		return errors.Wrap(err, "failed to create binding secret")
+		return nil, errors.Wrap(err, "failed to create binding secret")
 	}
 
 	cfServiceBinding.Status.Binding.Name = bindingSecret.Name
 
-	return nil
+	return bindingSecret, nil
+}
+
+func (r *UPSIBindingReconciler) reconcileSBServiceBinding(ctx context.Context, cfServiceBinding *korifiv1alpha1.CFServiceBinding, bindingSecret *corev1.Secret) (*servicebindingv1beta1.ServiceBinding, error) {
+	sbServiceBinding := r.toSBServiceBinding(cfServiceBinding)
+
+	_, err := controllerutil.CreateOrPatch(ctx, r.k8sClient, sbServiceBinding, func() error {
+		sbServiceBinding.Spec.Name = getSBServiceBindingName(cfServiceBinding)
+
+		secretType, hasType := bindingSecret.Data["type"]
+		if hasType && len(secretType) > 0 {
+			sbServiceBinding.Spec.Type = string(secretType)
+		}
+
+		secretProvider, hasProvider := bindingSecret.Data["provider"]
+		if hasProvider {
+			sbServiceBinding.Spec.Provider = string(secretProvider)
+		}
+		return controllerutil.SetControllerReference(cfServiceBinding, sbServiceBinding, r.scheme)
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return sbServiceBinding, nil
+}
+
+func (r *UPSIBindingReconciler) toSBServiceBinding(cfServiceBinding *korifiv1alpha1.CFServiceBinding) *servicebindingv1beta1.ServiceBinding {
```

Comment:
This function does not use the receiver `r` and therefore could be static and not a method. The same goes for its managed controller counterpart. 

It looks like converting the cfservicebinding to sbservicebinding is identical in both managed and upsi. Can we then extract this function to a common reusable place so that we have the code once?

Same goes for the function that checks sbservice binding readiness.